src/main.rs

```diff
@@ -21,14 +29,31 @@ fn main() -> anyhow::Result<()> {
                     continue;
                 }
 
-                items.push(format!(
-                    "{} = {}",
-                    entry.path().to_string_lossy().to_string(),
-                    repo.head()?.name().unwrap_or("None")
-                ));
+                let head = repo.head()?;
+                if let Some(head) = head.name() {
+                    if let Ok(upstream) = repo.branch_upstream_name(head) {
+                        if let Ok(remote) = repo.find_remote(
+                            // This is a rather ugly hack, but not sure how else to get the required name
+                            // doesn't seem to work with the full name such as `refs/remotes/origin/master`
+                            upstream
+                                .as_str()
+                                .unwrap_or("None")
```

Comment:
This should just become an `.unwrap()` so we don't get an unhelpful OOB panic later down this chain of methods.

Reply:
Right, making the changes